test/create_env_cluster_topics_test.go

```diff
@@ -76,30 +57,189 @@ func TestEnvClusterTopic(t *testing.T) {
 		defer terraform.Destroy(t, runOptions)
 		terraform.InitAndApply(t, runOptions)
 
-		var output string
+		// Fetch Terraform outputs
+		envID := terraform.Output(t, runOptions, "environment_id")
+		environmentName := terraform.Output(t, runOptions, "environment_name")
+		kafkaClusterID := terraform.Output(t, runOptions, "kafka_cluster_id")
+		bigQueryConnectorID := terraform.Output(t, runOptions, "bigquery_connector_id")
+		gcsSinkConnectorID := terraform.Output(t, runOptions, "connector_gcs_sink_connector_id")
+		topicServiceAccountID := terraform.Output(t, runOptions, "topic_service_account_id")
+		kafkaClusterBasicName := terraform.Output(t, runOptions, "kafka_cluster_basic_name")
+		kafkaTopicName1 := terraform.Output(t, runOptions, "kafka_topic_name_1")
+		kafkaTopicName2 := terraform.Output(t, runOptions, "kafka_topic_name_2")
+
+		// Confluent Login, environment, and Kafka cluster setup
+		os.Setenv("CONFLUENT_PLATFORM_PASSWORD", confluentCloudPassword)
+		os.Setenv("CONFLUENT_PLATFORM_EMAIL", confluentCloudEmail)
+		log.Printf("Setting Confluent environment: %s and cluster: %s...", envID, kafkaClusterID)
+		runCLICommand(t, "confluent", "env", "use", envID)
+		runCLICommand(t, "confluent", "kafka", "cluster", "use", kafkaClusterID)
+
+		// Validate Total ACL Provision
+		expectedTotalACLCount := 10
+		ActualTotalACLCount := validateTotalKafkaACLCountWithCLI(t)
+		assert.Equal(t, expectedTotalACLCount, ActualTotalACLCount, fmt.Sprintf(
+			"Expected %d ACLs to be created, but found %d", expectedTotalACLCount, ActualTotalACLCount))
+
+		// Validate individual ACL combinations count (ResourceType and Operation)
+		individualExpectedACLs := []struct {
+			ResourceType string
+			Operation    string
+			Count        int
+		}{
+			{"GROUP", "READ", 1},
```

Comment:
I'm not sure why we're validating in these kinds of aggregates rather than the specific attributes, given that we can easily range/loop over them. Rather than going through all these convolutions to group topics by their parameters, we should just validate the specific values. It would be simple to iterate since the data is all returned in the JSON anyways:

* principal
* permission
* operation
* resource_type
* resource_name
* pattern_type

```
  {
    "principal": "User:sa-xxp5zq",
    "permission": "ALLOW",
    "operation": "WRITE",
    "resource_type": "TOPIC",
    "resource_name": "dlq-lcc",
    "pattern_type": "PREFIXED"
  }
```

Reply:
While writing this logic I thought ACL was often designed with Resource Type and Operation Since the output was in array, i thought making a unique combination and count the combination will make more sense rather then defining the below data and matching the each object in the array.

I have a backlog ticket for this we gonna improve the logic 
https://linear.app/honestbank/issue/DEVOP-5630/add-more-test-coverage-in-the-terraform-confluent-kafka
```
 "principal": "User:sa-0n5m6q",
    "permission": "ALLOW",
    "operation": "CREATE",
    "resource_type": "TOPIC",
    "resource_name": "dlq-lcc",
    "pattern_type": "PREFIXED"
  ***,
  ***
    "principal": "User:sa-0n5m6q",
    "permission": "ALLOW",
    "operation": "DESCRIBE",
    "resource_type": "CLUSTER",
    "resource_name": "kafka-cluster",
    "pattern_type": "LITERAL"
  ***,
  ***
    "principal": "User:sa-0n5m6q",
    "permission": "ALLOW",
    "operation": "READ",
    "resource_type": "GROUP",
    "resource_name": "connect-lcc",
    "pattern_type": "PREFIXED"
  ***,
  ***
    "principal": "User:sa-0n5m6q",
    "permission": "ALLOW",
    "operation": "READ",
    "resource_type": "TOPIC",
    "resource_name": "squad_raw_service_example_1_entity",
    "pattern_type": "LITERAL"
  ***,
  ***
    "principal": "User:sa-0n5m6q",
    "permission": "ALLOW",
    "operation": "READ",
    "resource_type": "TOPIC",
    "resource_name": "squad_raw_service_example_2_entity",
    "pattern_type": "LITERAL"
  ***,
  ***
    "principal": "User:sa-0n5m6q",
    "permission": "ALLOW",
    "operation": "WRITE",
    "resource_type": "TOPIC",
    "resource_name": "dlq-lcc",
    "pattern_type": "PREFIXED"
  ***,
  ***
    "principal": "User:sa-oozjjj",
    "permission": "ALLOW",
    "operation": "READ",
    "resource_type": "TOPIC",
    "resource_name": "squad_raw_service_example_1_entity",
    "pattern_type": "LITERAL"
  ***,
  ***
    "principal": "User:sa-oozjjj",
    "permission": "ALLOW",
    "operation": "READ",
    "resource_type": "TOPIC",
    "resource_name": "squad_raw_service_example_2_entity",
    "pattern_type": "LITERAL"
  ***,
  ***
    "principal": "User:sa-oozjjj",
    "permission": "ALLOW",
    "operation": "WRITE",
    "resource_type": "TOPIC",
    "resource_name": "squad_raw_service_example_1_entity",
    "pattern_type": "LITERAL"
  ***,
  ***
    "principal": "User:sa-oozjjj",
    "permission": "ALLOW",
    "operation": "WRITE",
    "resource_type": "TOPIC",
    "resource_name": "squad_raw_service_example_2_entity",
    "pattern_type": "LITERAL"
  ***
]
~~~

Reply:
this is exactly what should be done - that's what testing is. Aggregating makes no sense - it means that the data can still be wrong.

Reply:
You've already typed up the data - just put it in the file and iterate/validate against it - don't see what the difficulty here?